File: Top_Interview_150_problems/2128-reverse-prefix-of-word/reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        for (int i = 0; i < word.length(); i++){
            if (word.charAt(i) == ch){
                StringBuilder sb = new StringBuilder();
                sb.append(word.substring(0, i+1));
                sb.reverse(); 
                sb.append(word.substring(i+1, word.length()));
                return sb.toString();
            }
        }
        return word;
    }
}

File: Top_Interview_150_problems/2128-reverse-prefix-of-word/reverse-prefix-of-word.kt
class Solution {
    fun reversePrefix(word: String, ch: Char): String {
        val strB = StringBuilder(word)

        val end = strB.indexOf(ch)

        if (end == -1){
            return word
        }

        var left = 0

        var right = end

        while(left < right) {
            val lS = strB.get(left)

            strB.set(left, strB.get(right))

            strB.set(right, lS)

            left++
            right--
        }

        return strB.toString()
    }
}